### ISSUE TITLE
PlayerOpEvent

### ISSUE THREAD

**Issue by literally1984** [hpfxd/PandaSpigot#163]
This is more of a suggestion than an issue but it would be nice if there was an event which was fired whenever a player is opped (whether that be from using /op or a plugin doing player.setOp().

**Comment by iamnoksio** [comment]
useless do it in plugin

**Comment by literally1984** [comment]
> useless do it in plugin

Right now I am using a BukkitRunnable, but I think that an actual event would be better.

**Comment by iamnoksio** [comment]
> > useless do it in plugin
> 
> Right now I am using a BukkitRunnable, but I think that an actual event would be better.

There's A LOT of better ways to do it without using a Runnable

**Comment by iamnoksio** [comment]
or there's something called: do it yourself

**Comment by andreasdc** [comment]
What are you trying to achieve?

**Comment by literally1984** [comment]
> or there's something called: do it yourself

You could say that every event Pandaspigot has backported is useless then, because one could do it themself.

**Comment by andreasdc** [comment]
> > or there's something called: do it yourself
> 
> You could say that every event Pandaspigot has backported is useless then, because one could do it themself.

Will you say what do you want to achieve or not

**Comment by literally1984** [comment]
> > > or there's something called: do it yourself
> > 
> > 
> > You could say that every event Pandaspigot has backported is useless then, because one could do it themself.
> 
> Will you say what do you want to achieve or not

What do you mean?

**Comment by andreasdc** [comment]
What do you need that feature for?

**Comment by literally1984** [comment]
> What do you need that feature for?

I would like to have the feature to cancel anybody save a few whitelisted players to get op. I'm already accomplishing this by using a BukkitRunnable and checking the op list every few seconds, but I feel as though an actual event would be better.

**Comment by andreasdc** [comment]
You can modify setting op using reflection and add whitelist check there.

**Comment by literally1984** [comment]
I'm sure I could do that, but it seems messy and an event would be much cleaner.

**Comment by iamnoksio** [comment]
> I'm sure I could do that, but it seems messy and an event would be much cleaner.

DUDE you buggin w your damn event, create it, put in in place and do a pull request

**Comment by iamnoksio** [comment]
> > What do you need that feature for?
> 
> I would like to have the feature to cancel anybody save a few whitelisted players from getting op. I'm already accomplishing this by using a BukkitRunnable and checking the op list every few seconds, but I feel as though an actual event would be better.

PlayerPreLoginEvent or some like that should be fine for what you've described

**Comment by uRyanxD** [comment]
I agree with iamnoksio, you can do it using the event he mentioned or simply make a plugin that blocks the execution of any command if the person is an operator and is not on the whitelist, so the person will not be able to do basically anything.
